src/psyclone/tests/psyclone_test_utils.py

```diff
@@ -109,29 +113,42 @@ class Compile(object):
     :type tmpdir: :py:class:`LocalPath`
     '''
 
-    # Whether or not we run tests requiring code compilation is picked-up
-    # from a command-line flag. (This is set-up in conftest.py.)
-    # pylint: disable=no-member
-    TEST_COMPILE = pytest.config.getoption("--compile")
-    # pylint: enable=no-member
-    # The following allows us to mark a test with @utils.COMPILE if it is
-    # only to be run when the --compile option is passed to py.test
-    COMPILE = pytest.mark.skipif(not TEST_COMPILE,
-                                 reason="Need --compile option to run")
-
-    # pylint: disable=no-member
-    TEST_COMPILE_OPENCL = pytest.config.getoption("--compileopencl")
-    # pylint: enable=no-member
-    COMPILE_OPENCL = pytest.mark.skipif(not TEST_COMPILE_OPENCL,
-                                        reason="Need --compileopencl "
-                                               "option to run")
+    # Class variable to store if compilation is enabled (--compile).
+    TEST_COMPILE = False
+
+    # Class variable to store if opencl compilation is enabled
+    # (--compileopencl).
+    TEST_COMPILE_OPENCL = False
+
+    # Class variable to store the chosen Fortran compiler (--f90, default
+    # gfortran).
+    F90 = "gfortran"
+
+    # Class variable to store the choosen f90 compiler flags (--f90flags).
+    F90FLAGS = ""
+
+    @staticmethod
+    def store_compilation_flags(config):
+        '''This function is called from the conftest session fixture
+        set_compilation flags. It sets the class variables related to
+        compilation based on the command line option.
+        :param config: The config object from pytest.
+        :type config: Instance of :py:class:`pytest.config.
+        '''
+        # Whether or not we run tests requiring code compilation is picked-up
+        # from a command-line flag. (This is set-up in conftest.py.)
+        Compile.TEST_COMPILE = config.getoption("--compile")
+        Compile.TEST_COMPILE_OPENCL = config.getoption("--compileopencl")
+        Compile.F90 = config.getoption("--f90")
+        Compile.F90FLAGS = config.getoption("--f90flags")
 
     def __init__(self, tmpdir=None):
         self._tmpdir = tmpdir
-        # pylint: disable=no-member
-        self._f90 = pytest.config.getoption("--f90")
-        self._f90flags = pytest.config.getoption("--f90flags")
-        # pylint: enable=no-member
+        # Take the compiler and compile flags from the statis variables.
+        # This allows a specific instance to use different compiler options
+        # which is used in some of the compilation tests.
```

Comment:
Should we therefore be careful to take copies, e.g.:

    self._f90 = Compile.F90[:]

or is this OK? i.e. we're not just storing a reference back to the static variable are we?

Reply:
Those are just string, and string copy does not create a reference.

Reply:
Aha!